src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollectionItem.cs

```diff
@@ -0,0 +1,162 @@
+                    return;
+
+                await newValue.InitTrackCollectionAsync();
+
+                // If any track in the parent collection has more than 2 artists,
+                // ALL track items should display their artist list, including this instance.
+                ShouldShowArtistList = newValue.Tracks.Any(x => x.TotalArtistItemsCount > 1) && newValue is IAlbum or IArtist;
+
+                lock (_lastKnownTrackArtistsCount)
+                {
+                    _lastKnownTrackArtistsCount.Clear();
+
+                    foreach (var item in newValue.Tracks)
+                    {
+                        _lastKnownTrackArtistsCount[item.Id] = item.TotalArtistItemsCount;
+                        AttachEvents(item);
+                    }
+
+                    AttachEvents(newValue);
+                }
+            }
+        }
+
+        private void AttachEvents(ITrackCollectionViewModel trackCollection)
+        {
+            trackCollection.TracksChanged += TrackCollection_TracksChanged;
+        }
+
+        private void DetachEvents(ITrackCollectionViewModel trackCollection)
+        {
+            trackCollection.TracksChanged -= TrackCollection_TracksChanged;
+        }
+
+        private void AttachEvents(ITrack track)
+        {
+            track.ArtistItemsChanged += Track_ArtistItemsChanged;
+        }
+
+        private void DetachEvents(ITrack track)
+        {
+            track.ArtistItemsChanged -= Track_ArtistItemsChanged;
+        }
+
+        private void TrackCollection_TracksChanged(object sender, IReadOnlyList<OwlCore.Events.CollectionChangedItem<ITrack>> addedItems, IReadOnlyList<OwlCore.Events.CollectionChangedItem<ITrack>> removedItems)
+        {
+            // Handle newly added/removed tracks
+            lock (_lastKnownTrackArtistsCount)
+            {
+                foreach (var item in removedItems)
+                    Guard.IsTrue(_lastKnownTrackArtistsCount.TryRemove(item.Data.Id, out _));
+
+                foreach (var item in addedItems)
+                    Guard.IsTrue(_lastKnownTrackArtistsCount.TryAdd(item.Data.Id, item.Data.TotalArtistItemsCount));
+
+                ShouldShowArtistList = _lastKnownTrackArtistsCount.Any(x => x.Value > 1) && _parentCollection is IAlbum or IArtist;
+            }
+        }
+
+        private void Track_ArtistItemsChanged(object sender, IReadOnlyList<OwlCore.Events.CollectionChangedItem<IArtistCollectionItem>> addedItems, IReadOnlyList<OwlCore.Events.CollectionChangedItem<IArtistCollectionItem>> removedItems)
+        {
+            // If any track in the parent collection has more than 2 artists,
+            // ALL track items should display their artist list, including this instance.
+            var track = (ITrack)sender;
+
+            // Fast path
+            if (track.TotalArtistItemsCount >= 2)
+            {
+                ShouldShowArtistList = _parentCollection is IAlbum or IArtist;
+                return;
```

Comment:
This return is premature. It still needs to adjust the `_lastKnownTrackArtistsCount` cache and run `ArtistNamesMetadata.ChangeCollection()`.